core-plugins/src/test/java/co/cask/hydrator/plugin/batch/action/OracleExportActionTest.java

```diff
@@ -0,0 +1,138 @@
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.util.EnumSet;
+
+/**
+ * Test for {@link OracleExportAction}
+ */
+public class OracleExportActionTest extends ETLBatchTestBase {
+  @ClassRule
+  public static final TemporaryFolder TEMP_FOLDER = new TemporaryFolder();
+  private static final String USER = "test";
+  private static final String PASSPHRASE = "password";
+  private static final String[] SHELL_ARGS = {"/bin/sh", "-i", "-l"};
+  private static SshServer sshServer;
+  private static MiniDFSCluster dfsCluster;
+  private static FileSystem fileSystem;
+
+  @BeforeClass
+  public static void buildMiniDFS() throws Exception {
+    // Setup MiniDFSCluster
+    File baseDir = TEMP_FOLDER.newFolder();
+    Configuration conf = new Configuration();
+    conf.set(MiniDFSCluster.HDFS_MINIDFS_BASEDIR, baseDir.getAbsolutePath());
+    dfsCluster = new MiniDFSCluster.Builder(conf).build();
+    dfsCluster.waitActive();
+    fileSystem = FileSystem.newInstance(conf);
+    sshServer = setupSSHDaemon();
+    sshServer.start();
+  }
+
+  @AfterClass
+  public static void cleanupMiniDFS() throws Exception {
+    // Shutdown MiniDFSCluster
+    dfsCluster.shutdown();
+    if (sshServer != null) {
+      sshServer.stop();
+    }
+  }
+
+  private static SshServer setupSSHDaemon() throws Exception {
+    SshServer sshServer = SshServer.setUpDefaultServer();
+    SimpleGeneratorHostKeyProvider simpleGeneratorHostKeyProvider = new SimpleGeneratorHostKeyProvider();
+    simpleGeneratorHostKeyProvider.setAlgorithm(KeyUtils.RSA_ALGORITHM);
+    sshServer.setKeyPairProvider(simpleGeneratorHostKeyProvider);
+    sshServer.setPasswordAuthenticator(new PasswordAuthenticator() {
+      @Override
+      public boolean authenticate(String username, String password, ServerSession session) {
+        return USER.equals(username) && PASSPHRASE.equals(password);
+      }
+    });
+    sshServer.setShellFactory(new ProcessShellFactory(SHELL_ARGS));
+    sshServer.setCommandFactory(EchoCommandFactory.INSTANCE);
+    return sshServer;
+  }
+
+  @Test(expected = SSHAuthenticationException.class)
+  public void testCheckWrongArgumentForServerPassword() throws Exception {
+    OracleExportAction.OracleExportActionConfig oracleExportActionConfig =
+      new OracleExportAction.OracleExportActionConfig(sshServer.getHost(),
+                                                      sshServer.getPort(), USER, "wrongPassword", "system",
+                                                      "cask", "home", "sid", "/tmp/junit.csv",
+                                                      "select * from test;", "/tmp", "csv");
+    OracleExportAction oracleExportAction = new OracleExportAction(oracleExportActionConfig);
+    MockActionContext mockActionContext = new MockActionContext();
+      oracleExportAction.run(mockActionContext);
+      Assert.fail();
```

Comment:
Don't need this. The test will fail if ``SSHAuthenticationException`` does not occur.